packages/core/src/shared/extensions/ssh.ts

```diff
@@ -130,27 +130,34 @@ export class RemoteSshSettings extends Settings.define('remote.SSH', remoteSshTy
     }
 }
 
+/**
+ * Test a SSH connection over SSM.
+ * @param ProcessClass given process to test the connection within.
+ * @param hostname
+ * @param sshPath
+ * @param user
+ * @param session SSM session credentials. These cannot be reused, so it may be required to create a seperate session for the test connection.
+ * @returns
+ */
 export async function testSshConnection(
     ProcessClass: typeof ChildProcess,
     hostname: string,
     sshPath: string,
     user: string,
     session: SSM.StartSessionResponse
 ): Promise<ChildProcessResult | never> {
+    const env = { SESSION_ID: session.SessionId, STREAM_URL: session.StreamUrl, TOKEN: session.TokenValue }
+    const process = new ProcessClass(sshPath, ['-T', `${user}@${hostname}`, 'echo "test connection succeeded" && exit'])
     try {
-        const env = { SESSION_ID: session.SessionId, STREAM_URL: session.StreamUrl, TOKEN: session.TokenValue }
-        const result = await new ProcessClass(sshPath, [
-            '-T',
-            `${user}@${hostname}`,
-            'echo "test connection succeeded" && exit',
-        ]).run({
+        return await process.run({
             spawnOptions: {
                 env,
             },
         })
-        return result
     } catch (error) {
-        throw new SshError('SSH connection test failed', { cause: error as Error })
+        throw new SshError(process.result()?.stderr ?? 'An unknown error occurred when testing the connection', {
+            cause: error as Error,
```

Comment:
Will this error serialize to a string correctly? or do we need (error as Error).message?

Reply:
`cause` field is always a `Error` subclass. The commands2.ts wrapper will get its fields for telemetry.

Reply:
It can be a little verbose, but it does serialize properly. I added an example to the PR description.